exporter/kafkaexporter/jaeger_marshaler_debug.go

```diff
@@ -0,0 +1,137 @@
+package kafkaexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter"
+
+// Similar to jaegerMarshaler except we log details of spans greater than producer maxMessageBytes. When doing otel
+// upgrades pull in updates from jaeger_mashaler.go:Marshal function
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	jaegerproto "github.com/jaegertracing/jaeger/model"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/multierr"
+
+	jaegertranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+)
+
+const (
+	maximumRecordOverhead   = 5*binary.MaxVarintLen32 + binary.MaxVarintLen64 + 1
+	producerMessageOverhead = 26 // the metadata overhead of CRC, flags, etc.
+)
+
+type jaegerMarshalerDebug struct {
```

Comment:
I would have added this fix in the jaegerSpanMarshaler itself.

Reply:
mmh that's modifying an original file. i decided to separate them since this one requires config that's not available at factory creation time like the producer maxmessagebytes. either way we would have needed a .modified.go file.